### PR TITLE
eth: eth_mcux: wait in send for the packet to be sent [REVERT ME]

### PR DESCRIPTION
**Pull request by tgorochowik**
Wait in the send callback for the packet to be actually sent.

After this change, only one TX packet will be handled at once.
This is needed because of the way the TX packets are currently handled
in L2 after this PR: #12563

This is similar to what #13167 did for the SAM GMAC on SAM E-70.

Without this, packet time-stamping does not work with the current stack.

This commit is minimalistic on purpose to make it easily revertible when
the network stack is able to properly handle DMA drivers for TX packets
again.

Signed-off-by: Tomasz Gorochowik <tgorochowik@antmicro.com>

### CODE REVIEW

**Review by jukkar — APPROVED** [review]

**Review by tbursztyka — APPROVED** [review]

File: drivers/ethernet/Kconfig.mcux [hunk 1]
```diff
@@ -46,7 +46,7 @@ config ETH_MCUX_TX_BUFFERS
 	int "Number of MCUX TX buffers"
 	depends on ETH_MCUX
 	default 1
-	range 1 16
+	range 1 1
```
Comment (pfalcon) [inline drivers/ethernet/Kconfig.mcux]: Is this change really required in this PR? Or is it along the lines of "don't promise false things"?

**Review by pfalcon — APPROVED** [review]
Hmm, I didn't think we have any async handling in eth_mcux... Sounds good to scale it down, yeah.

**Review by agansari — APPROVED** [review]
I'm approving this because it has a [REVERT ME] label.

**Review by nashif — APPROVED** [review]
